internal/http/services/owncloud/ocdav/ocdav_blackbox_test.go

```diff
@@ -203,6 +204,51 @@ var _ = Describe("ocdav", func() {
 			Entry("at the /dav/public-files endpoint for a folder", "/dav/public-files/tokenforfolder", "/public/tokenforfolder", http.StatusInternalServerError),
 		)
 
+		DescribeTable("HandleMkcol",
+			func(endpoint string, expectedPathPrefix string, expectedStatPath string, expectedStatus int) {
+
+				client.On("ListStorageSpaces", mock.Anything, mock.MatchedBy(func(req *cs3storageprovider.ListStorageSpacesRequest) bool {
+					p := string(req.Opaque.Map["path"].Value)
+					return p == "/" || strings.HasPrefix(p, expectedPathPrefix)
+				})).Return(nil, fmt.Errorf("unexpected io error"))
+
+				// path based requests need to check if the resource already exists
+				client.On("Stat", mock.Anything, mock.MatchedBy(func(req *cs3storageprovider.StatRequest) bool {
+					return req.Ref.Path == expectedStatPath
+				})).Return(&cs3storageprovider.StatResponse{
+					Status: status.NewNotFound(ctx, "not found"),
+				}, nil)
+
+				// the spaces endpoint omits the list storage spaces call, it directly executes the delete call
```

Comment:
this comment seems wrong

Reply:
but it is correct. when using /dav/spaces we can forward the request directly to the gateway, because we know the storageid and spaceid. the gateway will query the storage registry and forward accordingly.

Path based requests need to figure out the spaceid by listing the storage spaces for the requested path, before making the actual request.